src/connectors/autocomplete/connectAutocomplete.js

```diff
@@ -0,0 +1,153 @@
+ * @property {string[]} [indices = []] Name of the others indices to search into.
+ * @property {boolean} [escapeHits = false] If true, escape HTML tags from `hits[i]._highlightResult`.
+ */
+
+/**
+ * **Autocomplete** connector provides the logic to build a widget that will give the user the ability to search into multiple indices.
+ *
+ * This connector provides a `refine()` function to search for a query and a `currentRefinement` as the current query used to search.
+ *
+ * THere's a complete example available on how to write a custom **Autocomplete** widget:
+ * [autocomplete.js](https://github.com/algolia/instantsearch.js/blob/develop/dev/app/custom-widgets/jquery/autocomplete.js)
+ * @type {Connector}
+ * @param {function(AutocompleteRenderingOptions, boolean)} renderFn Rendering function for the custom **Autocomplete** widget.
+ * @param {function} unmountFn Unmount function called when the widget is disposed.
+ * @return {function(CustomAutocompleteWidgetOptions)} Re-usable widget factory for a custom **Autocomplete** widget.
+ */
+export default function connectAutocomplete(renderFn, unmountFn) {
+  checkRendering(renderFn, usage);
+
+  return (widgetParams = {}) => {
+    const { indices } = widgetParams;
+
+    if (indices && !Array.isArray(indices)) {
+      throw new Error(usage);
+    }
+
+    return {
+      getConfiguration() {
+        return widgetParams.escapeHits ? tagConfig : undefined;
+      },
+
+      init({ instantSearchInstance, helper }) {
+        this._refine = this.refine(helper);
+
+        this.indices = [
+          {
+            helper,
+            label: 'primary',
+            index: helper.getIndex(),
+            results: undefined,
+          },
+        ];
+
+        // add additionnal indices into `this.indices`
+        if (indices && Array.isArray(indices)) {
+          indices.forEach(({ label, value }) => {
+            const derivedHelper = helper.derive(searchParameters =>
+              searchParameters.setIndex(value)
+            );
+
+            this.indices.push({
+              label,
+              index: value,
+              helper: derivedHelper,
+              results: undefined,
+            });
+
+            // update results then trigger render after a search from any helper
+            derivedHelper.on('result', results =>
+              this.saveResults({ results, index: value })
+            );
+          });
+        }
+
+        this.instantSearchInstance = instantSearchInstance;
+        this.renderWithAllIndices({ isFirstRendering: true });
+      },
+
+      saveResults({ results, index }) {
+        const derivedIndex = this.indices.find(i => i.index === index);
+
+        if (
+          widgetParams.escapeHits &&
+          results.hits &&
+          results.hits.length > 0
+        ) {
+          results.hits = escapeHits(results.hits);
+        }
+
+        derivedIndex.results = results;
+        derivedIndex.hits = results ? results.hits : undefined;
```

Comment:
Why don't we use an empty array as default value? It means that we always need to check either the `results` or the `hits` in order to render. With a default value we can always loop on the hits.

Reply:
Good point 👍 I still prefer the empty array but indeed it's the same. But it's only my opinion, we can keep it like that.

Reply:
I'd prefer a single type and then be able to check on the number of hits rather than if hits is there or not

Reply:
Alright let's go for it 🚢